_oasis

```diff
@@ -35,84 +35,78 @@ Library tcpip_xen
 
 Library ethif
   CompiledObject:     best
-  Path:               lib
+  Path:               lib/ethif
   Findlibparent:      tcpip
   Findlibname:        ethif
-  Modules:            Ethif
-  BuildDepends:       tcpip,io-page,mirage-types,ipaddr,cstruct,lwt
+  Modules:            Ethif, Ethif_wire, Ethif_parse, Ethif_print
+  BuildDepends:       mirage-types,ipaddr,cstruct,lwt,mirage-profile,result,rresult
 
 Library arpv4
   CompiledObject:     best
-  Path:               lib
+  Path:               lib/arpv4
   Findlibparent:      tcpip
   Findlibname:        arpv4
-  Modules:            Arpv4, Arpv4_wire
-  BuildDepends:       tcpip,io-page,mirage-types,ipaddr,cstruct,cstruct.ppx,lwt
+  Modules:            Arpv4, Arpv4_wire, Arpv4_parse, Arpv4_print
+  BuildDepends:       mirage-types,logs,ipaddr,cstruct,cstruct.ppx,lwt,
```

Comment:
this would probably requires a XMETArequires line to not depend on `ppx` at runtime. Note: I'm not totally sure, so worth checking with @djs55 who had this issue recently.